File: app/src/main/java/com/returnnotfound/stain/di/module/ActivityBuilder.java
package com.returnnotfound.stain.di.module;

import com.returnnotfound.stain.di.ActivityScope;
import com.returnnotfound.stain.ui.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityBuilder {

  @ContributesAndroidInjector(modules = FragmentBuilder.class)
  @ActivityScope
  abstract MainActivity contributeMainActivity();
}

File: app/src/main/java/com/returnnotfound/stain/di/module/ActivityBuilder.kt
package com.returnnotfound.stain.di.module

import com.returnnotfound.stain.di.ActivityScope
import com.returnnotfound.stain.presentation.ui.MainActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityBuilder {
  @ContributesAndroidInjector(modules = [FragmentBuilder::class])
  @ActivityScope
  abstract fun contributeMainActivity(): MainActivity
}